src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/AssessmentInfoFactory.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.factory
+
+import io.github.bluegroundltd.kfactory.Factory
+import io.github.bluegroundltd.kfactory.Yielded
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.oasyscontext.AssessmentInfo
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomDateTimeBefore
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomInt
+import java.time.LocalDateTime
+
+abstract class AssessmentInfoFactory<T : AssessmentInfo> : Factory<T> {
+  protected var assessmentId: Yielded<Long> = { randomInt(1000, 10000).toLong() }
+  protected var assessmentType: Yielded<String> = { "LAYER3" }
+  protected var dateCompleted: Yielded<LocalDateTime?> = { null }
```

Comment:
maybe we'll want a trait and a test to show how we handle response which both contain and lack the `dateCompleted` property?

Reply:
dateCompleted isn't actually used anywhere at the moment - have I missed some functionality?

Reply:
I think we will want to surface some additional metadata on the assessment so that we can display information like:

- Assessment state: e.g. "Completed"
- Date: date completed if present else date started

But for now I guess as long as it doesn't blow up if the property is missing is enough?

Reply:
Happy to add this in now if you think we'll need it sooner than later - up to you?

Reply:
Have added this